### DISCUSSION TITLE
Loki Annotations does not support duration

### DISCUSSION THREAD

**Discussion by justaverylee** [grafana/grafana#69105] [Dashboards, Panels & Visualizations]
# TLDR
Some annotations provider datasources allow you to specify both a start and end time, others (like loki) don't. I would like to add that feature.

# Context
We used to store an index of our changes in an elastic cluster and plot them as annotations on our dashboards. This allowed us to use one field as the start time and another as the end time and grafana would plot these as a range.
We've now migrated to Logql+Loki and I'm now left without the ability to present a range of times on my annotations and can only plot annotations at the "event time"

# The Ask
It would be great to add an option to set "start time" and or "event time" based on a field (instead of event time) to the annotations screen of the loki plugin.

So the technical problem I understand comes from the fact that when querying if the event time is significantly different from the end time you'd only get annotations which have the event time within your dashboard time range. Suggestions on how to handle this case would be appreciated

**Comment by justaverylee** [comment]
Edit annotations screen for Loki vs Elastic
<img width="592" alt="image" src="https://github.com/grafana/grafana/assets/14918261/a716ec78-82a7-4d20-b394-dabdd5e1a848">
<img width="725" alt="image" src="https://github.com/grafana/grafana/assets/14918261/e296c076-13af-4c7f-aeaf-4d9d612c1bc5">
Specifically please see the "Time" and "Time End" fields

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82629. Feel free to continue the discussion around this there. Thank you!
